syntax = "proto3";

package envserver;

service EnvService {
  rpc Reset (ResetRequest) returns (ResetResponse);
  rpc Step (StepRequest) returns (StepResponse);
}

message ResetRequest {
  string env = 1;
}

message ResetResponse {
  string session_id = 1;
  repeated float observation = 2;
}

message StepRequest {
  string session_id = 1;
  int32 action = 2;
}

message StepResponse {
  repeated float observation = 1;
  float reward = 2;
  bool done = 3;
  map<string, string> info = 4;
}
